### PR TITLE
Make src/sidebar/util typecheck with `noImplicitAny`

### PR DESCRIPTION
**Pull request by robertknight**
This is largely straightforward. The largest change is elimination of an unnecessary helper function in `search-filter.js` to save on the amount of type annotations. See comments.

### CODE REVIEW

File: src/sidebar/util/search-filter.js [hunk 1]
```diff
@@ -91,25 +94,25 @@ function tokenize(searchText) {
...
-      return (obj[key] = [data]);
-    }
-  };
+  /** @param {string} field */
+  const backendFilter = field => (field === 'tag' ? 'tags' : field);
 
   if (searchText) {
     const terms = tokenize(searchText);
     for (const term of terms) {
-      let [filter, data] = splitTerm(term);
```
Comment (robertknight) [inline src/sidebar/util/search-filter.js:107]: `term` here is a string like `title:foobar` and `splitTerm` splits it into the field name and value parts. I changed `filter` to `field` as I thought that was a more obvious name.

File: src/sidebar/util/search-filter.js [hunk 2]
```diff
@@ -91,25 +94,25 @@ function tokenize(searchText) {
 export function toObject(searchText) {
   /** @type {Record<string,string[]>} */
   const obj = {};
-  const backendFilter = f => (f === 'tag' ? 'tags' : f);
 
-  const addToObj = (key, data) => {
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/util/search-filter.js:96]: Hah. I had to re-read that four times before realizing (I think?!) that _Login_ should be _Logic_.

File: src/sidebar/util/search-filter.js (hunk 1 above)
Comment (lyzadanger) [reply src/sidebar/util/search-filter.js:107]: I agree.

**Review by lyzadanger — APPROVED** [review]
LGTM.

**Pull request merged**